File: AdventJS/ejercicio_7.py
class Orden_Regalos:
    def fix_packages(self, packages):
        if len(packages) == 0:
            return ''
        
        lista_tmp = []
        parentesis = []
        for letter in packages:
            lista_tmp.append(letter)
            if lista_tmp.count('(') == 1 and letter == '(':
                lista_tmp = []
            elif letter == ')':
                parentesis = lista_tmp[:-1]
                tmp, tmx = '('+''.join(lista_tmp), ''.join(parentesis)
                packages = packages.replace(tmp, tmx[::-1])
                break

        if '(' in packages:
            return self.fix_packages(packages)
    
        return packages


packages = '(abc(def(ghi)))'
obj = Orden_Regalos()
print(obj.fix_packages(packages))

File: AdventJS/ejercicio_7.js
function fixPackages(packages) {
    if(packages.length == 0)
      return '';
      
    var lista_tmp = [], letras = [], tmp ='';
    packages.split('').forEach(e => {
        lista_tmp.push(e);
      if(lista_tmp.includes('(') && e == '(')
          lista_tmp = [];
      else if(lista_tmp.length > 1 && e == ')'){
          letras = lista_tmp.slice(0, -1).reverse().join('');
          tmp = '('+lista_tmp.join('');
          packages = packages.replace(tmp, letras);
      }
    });
    
    
    if(packages.includes('('))
        return fixPackages(packages);

    return packages;
    
  }